lib/child_process.js

```diff
@@ -50,7 +50,20 @@ exports.fork = function(modulePath /*, args, options*/) {
 
   args = execArgv.concat([modulePath], args);
 
-  if (!Array.isArray(options.stdio)) {
+  if (typeof options.stdio === 'string') {
+    const stdioStringToArray = (option) => {
+      return [option, option, option, 'ipc'];
+    };
+    switch (options.stdio) {
+      case 'ignore':
+      case 'pipe':
+      case 'inherit':
+        options.stdio = stdioStringToArray(options.stdio);
+        break;
+      default:
+        throw new TypeError(`Incorrect type of stdio option: ${options.stdio}`);
```

Comment:
> `git grep TypeError lib`

Funny, why does `git` (re-?)implement `grep`?

> So at this point, we have one example of a format string inputting an invalid argument into the type error message, and this would be the second, so now I'm convinced.

These are the positive results when not restricting to `TypeError`s (or is there any particular reason you only looked for those?): https://gist.github.com/addaleax/cd3d3e5a6c7d752eaa1d15396e80608e (“only” 12 but still.)

But why does it really matter whether there are precedents for this? I can’t really see any situation in which this is harmful, and I can see cases in which echoing the input back might be helpful (very simple example: typos).